src/librbd/api/Group.cc

```diff
@@ -485,8 +253,15 @@ int GroupSnapshot_to_group_snap_info2(
   group_snap->id = cls_group_snap.id;
   group_snap->name = cls_group_snap.name;
   group_snap->state = static_cast<group_snap_state_t>(cls_group_snap.state);
-  group_snap->namespace_type = RBD_GROUP_SNAP_NAMESPACE_TYPE_USER;
-  if (!image_snaps.empty()) {
+  if (std::holds_alternative<cls::rbd::GroupSnapshotNamespaceMirror>(
+	cls_group_snap.snapshot_namespace)) {
+    group_snap->namespace_type = RBD_GROUP_SNAP_NAMESPACE_TYPE_MIRROR;
```

Comment:
I'd suggest following what is done for image snapshots, even if it's more verbose: introduce `get_group_snap_namespace_type()` modeled after `get_snap_namespace_type()` in `src/cls/rbd/cls_rbd_types.cc` and do
```
    group_snap->namespace_type = static_cast<group_snap_namespace_type_t>(
      cls::rbd::get_group_snap_namespace_type(cls_group_snap.snapshot_namespace));
```
here.

Reply:
`get_group_snap_namespace_type() ` was already implemented. Just used it as you suggested.